Expensify/App
# Allow e.com to link straight to e.cash and share sessions

### Fixed Issues
Done along with: https://github.com/Expensify/Web-Expensify/pull/31302
Fixes https://github.com/Expensify/Expensify/issues/162554

### Tests / QA
1. Be sure you're signed out of e.cash
2. Sign into e.com with a brand new account on a domain WITHOUT two-factor authentication
3. Complete the inbox task "What will you use Expensify for? Choose as many as you'd like!" and ONLY choose the expensify card
4. Verify that you are automatically logged into e.cash and are seeing the page to create a new workspace

Now, test it out with 2FA

1. Be sure you're signed out of e.cash
2. Sign into e.com with a brand new account on a domain WITH two-factor authentication enabled
3. Complete the inbox task "What will you use Expensify for? Choose as many as you'd like!" and ONLY choose the expensify card
4. Verify that you are redirected to e.cash to ever a 2FA code
5. After the code is entered, verify you are automatically logged into e.cash and are seeing the page to create a new workspace

### Tested On
No other platforms need to be tested because this flow only works on web (the inbox task isn't shown on mobile at all)

### Screenshots


#### Web

![image](https://user-images.githubusercontent.com/1228807/124313390-0d1e4c00-db2e-11eb-9c85-abe239b54817.png)

cc @marcaaron @TomatoToaster
